packages/core/src/indexing-store/utils/filter.ts

```diff
@@ -0,0 +1,153 @@
+import type { Schema } from "@/schema/types.js";
+import { isBaseColumn, isEnumColumn } from "@/schema/utils.js";
+import type { ComparisonOperatorExpression } from "kysely";
+import type { Table, WhereInput } from "../store.js";
+import { encodeValue } from "./encoding.js";
+
+const filterValidityMap = {
+  boolean: {
+    singular: ["equals", "not", "in", "notIn"],
+    list: ["equals", "not", "has", "notHas"],
+  },
+  string: {
```

Comment:
What are enums encoded as?

Reply:
Ended up treating them the exact same way as strings.